webproject/src/main/java/com/google/sps/data/Room.java

```diff
@@ -46,12 +53,36 @@ private Room (Builder builder) {
     public double getMinPrice() {return minPrice;}
 
     public double getDeliveryFee() {return deliveryFee;}
+
     public List<String> getUsers() {return users;}
 
     public String getCreator() {return creator;}
 
-    public void addUser(String userId) {
-        this.users.add(userId);
+    public double getOrdersValue() {return ordersValue;}
+
+    public void addUser(String userEmail) {
+        this.users.add(userEmail);
+    }
+
+    public void addOrdersValue(double orderPrice) {
+        this.ordersValue += orderPrice;
+    }
+
+    public void save() throws IOException {
+        String roomJson = gson.toJson(this);
+        Firebase.sendRequest("https://summer20-sps-47.firebaseio.com/rooms.json", "POST", roomJson);
+    }
+
+    public void save(String roomId) throws IOException {
```

Comment:
there should be only 1 save() function without param
the roomId can be set as null initially in the builder, and the save() function should check, if it is null, it should treat it as "insert" whereas if its not null, it should treat it as "update"